Structure_Engine/Query/ReinforcementLayout.cs

```diff
@@ -0,0 +1,207 @@
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Structure.SectionProperties;
+using BH.oM.Structure.SectionProperties.Reinforcement;
+using BH.oM.Structure.Elements;
+using BH.oM.Spatial.Layouts;
+using BH.oM.Geometry;
+using BH.Engine.Spatial;
+using BH.Engine.Geometry;
+using BH.oM.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Gets the LongitudinalReinforcement positions in the ConcreteSection as a list of Points.")]
+        [Input("section", "The concrete section to return all points from.")]
+        [Input("position", "Position along the section to extract reinforcement. A negative value will return all reinforcement.")]
+        [Output("points", "The positions of the LongitudinalReinforcement.")]
+        public static List<Point> ReinforcementLayout(ConcreteSection section, double position = -1)
+        {
+            //Extract Longitudinal reinforcement
+            List<LongitudinalReinforcement> longReif = section.LongitudinalReinforcement();
+
+            //No longitudinal reinforcement available
+            if (longReif.Count == 0)
+                return new List<Point>();
+
+            List<ICurve> outerProfileEdges;
+            List<ICurve> innerProfileEdges;
+
+            ExtractInnerAndOuterEdges(section, out outerProfileEdges, out innerProfileEdges);
+
+            //Need at least one external edge curve
+            if (outerProfileEdges.Count == 0)
+                return new List<Point>();
+
+            //TODO: include stirups for offset distance
```

Comment:
Would add an issue instead of a TODO comment

Reply:
Issue raised:
https://github.com/BHoM/BHoM_Engine/issues/1770